enhancements/baremetal/coreos-image-in-release.md

```diff
@@ -0,0 +1,258 @@
+  - "@dhellmann"
+  - "@sdodson"
+  - "@LorbusChris"
+approvers:
+  - "@hardys"
+  - "@aravindhp"
+creation-date: 2021-09-22
+last-updated: 2021-09-22
+status: implementable
+see-also:
+  - "/enhancements/coreos-bootimages.md"
+---
+
+# Include the CoreOS image in the release for baremetal
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The baremetal platform is switching from a QCOW2 format for its CoreOS image to
+an ISO. To ensure that existing disconnected clusters can update to this, the
+ISO image will be included in the release payload. This will be balanced out by
+removing the RHEL image currently shipped in the release payload. There are
+many other nice benefits, such that after future upgrades clusters will
+provision new machines with the latest CoreOS, without requiring an extra
+reboot.
+
+## Motivation
+
+Currently, the deploy disk image (i.e. the image running IPA -
+ironic-python-agent) is a RHEL kernel plus initrd that is installed (from an
+RPM) into the `ironic-ipa-downloader` container image, which in turn is part of
+the OpenShift release payload. When the metal3 Pod starts up, the disk image is
+copied from the container to a HostPath volume whence it is available to
+Ironic.
+
+The provisioning OS disk image is a separate CoreOS QCOW2 image. The URL for
+this is known by the installer. It points to the cloud by default and may be
+customised by the user to allow disconnected installs. The URL is stored in the
+Provisioning CR at install time and never updated automatically. The image
+itself is downloaded once and permanently cached on all of the master nodes.
+Never updating the image is tolerable because, upon booting, the CoreOS image
+will update itself to the version matching the cluster it is to join. It
+remains suboptimal because new Machines will take longer and longer (and more
+and more bandwidth) to join as the cluster ages. This issue exists on all
+platforms, and is the subject of a [long-standing enhancement
+proposal](https://github.com/openshift/enhancements/pull/201). Other issues
+specific to the baremetal platform are that boot times for bare metal servers
+can be very long (and therefore the reboot is costly), and that support for
+particular hardware may theoretically require a particular version of CoreOS.
+
+We are changing the deploy disk image to use CoreOS. This takes the form of
+both an ISO (for hosts that can use virtualmedia) and of a kernel + initrd +
+rootfs (for hosts that use PXE). When upgrading an existing disconnected
+cluster, we have no way to acquire these images without the user manually
+intervening to mirror them.
```

Comment:
Again here I'd maybe say "We are changing to use the same CoreOS based images provided for UPI deployments" or similar, just so it's clear we're aligning the flows vs generating any net-new artifacts.